modules/h2/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteTrace.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.processors.query.h2;
+
+/**
+ * Tracing object for SQL queries.
+ */
+public class IgniteTrace {
```

Comment:
why do we need one more holder for metrics? There is already QueryMemTracker. Since it is created on per query basis, we could just simply extend it with some query description. IMO it should be enough

Reply:
I wouldn't say I am absolutely agree with your point, but ok :)